store/primary/primary.go

```diff
@@ -72,25 +62,18 @@ func (s *rtStorage) Put(c cid.Cid, providerID peer.ID, pieceID cid.Cid) error {
 // PutCheck stores a provider-piece entry for a CID if the entry is not already
 // stored.  New entries are added to the entries that are already there.
 // Returns true if a new entry was added to the cache.
+//
+// Only rotate one cache at a time. This may leave older entries in other
```

Comment:
After reading the code (and understanding what you mentioned in our sync) I think the benefits of this implementation outgrow the overhead of leaving some outdated values behind memory-wise. 

What is less clear to me is if by adding an additional cache for entries, and consequently having to access two/three trees to get a value (vs. one/two in the implementation) we are adding a time overhead. Did you have a chance to test this?

Worst case scenario, we can configure the use of internValues as an option (that way we can have caches with and without entry deduplication) in case we want to compare both approaches.

Reply:
The overhead to having additional trees, as compared to additional branches in the same tree, is insignificant.  I had in fact looked at this and considered prefixing different types of data with a different byte to make separate branches, but that did not save anything.